tabular/src/autogluon/tabular/predictor/interpretable_predictor.py

```diff
@@ -0,0 +1,98 @@
+
+
+# TODO: Forcibly disable bagging, stacking, calibration, and non-interpretable models
+class InterpretableTabularPredictor(TabularPredictor):
+    def fit(self,
+            train_data,
+            tuning_data=None,
+            time_limit=None,
+            *,
+            presets='interpretable',
+            fit_weighted_ensemble=False,
+            calibrate=False,
+            **kwargs):
+        logger.log(30, f'EXPERIMENTAL WARNING: Fitting {self.__class__.__name__}\n'
+                       f'\tThis class is experimental and could be removed without warning in a future release.\n'
+                       f'\tTo avoid confusing results, please only provide numeric features.\n'
+                       f'\tCategorical, text, and datetime features will result in confusing rules that are hard to interpret.')
+
+        return super().fit(
+            train_data=train_data,
+            tuning_data=tuning_data,
+            time_limit=time_limit,
+            presets=presets,
+            fit_weighted_ensemble=fit_weighted_ensemble,
+            **kwargs,
+        )
+
+    def interpretable_models_summary(self, verbosity=0):
+        '''Summary of fitted interpretable models along with their corresponding complexities
+        '''
+        d = self.fit_summary(verbosity=verbosity)
+        summaries = pd.DataFrame.from_dict(d)
+
+        complexities = []
+        info = self.info()
+        for i in range(summaries.shape[0]):
+            model_name = summaries.index.values[i]
+            complexities.append(info['model_info'][model_name].get('complexity', np.nan))
+        summaries.insert(2, 'complexity', complexities)
+        summaries = summaries[~pd.isna(summaries.complexity)]  # remove non-interpretable models
+        return summaries.sort_values(by=['model_performance', 'complexity'], ascending=[False, True])
+
+    def print_interpretable_rules(self, complexity_threshold: int = 10, model_name: str = None):
+        """
+        Print the rules of the highest performing model below the complexity threshold.
+
+        Parameters
+        ----------
+        complexity_threshold : int, default=10
+            Threshold for complexity (number of rules) of fitted models to show.
+            If not model complexity is below this threshold, prints the model with the lowest complexity.
+        model_name : str,  default=None
+            Optionally print rules for a particular model, ignoring the complexity threshold.
+        """
+        if model_name is None:
+            summaries = self.interpretable_models_summary()
+            summaries_filtered = summaries[summaries.complexity <= complexity_threshold]
+            if summaries_filtered.shape[0] == 0:
+                summaries_filtered = summaries
+            model_name = summaries_filtered.index.values[0]  # best model is at top
+        agmodel = self._trainer.load_model(model_name)
+        imodel = agmodel.model
+        print(imodel)
```

Comment:
Use logging; print won't get into a file if filehandler is used

Reply:
keeping as print, we use print for leaderboard, and this is similar